datatypes/src/primitives/query_rectangle.rs

```diff
@@ -16,6 +16,7 @@ pub struct QueryRectangle<SpatialBounds: AxisAlignedRectangle> {
 pub type VectorQueryRectangle = QueryRectangle<BoundingBox2D>;
 pub type RasterQueryRectangle = QueryRectangle<SpatialPartition2D>;
 pub type PlotQueryRectangle = QueryRectangle<BoundingBox2D>;
+pub type MachineLearningQueryRectangle = QueryRectangle<BoundingBox2D>;
```

Comment:
just use the vectorqueryrectangle in your code, or define the type in the machine learning module in the services crate